Dockerfile

```diff
@@ -1,15 +1,19 @@
-FROM ruby:2.6-slim-stretch
+FROM ruby:2.7-slim
 WORKDIR /app
 
 RUN apt-get update && \
     apt-get install --no-install-recommends -y \
     build-essential \
     libpq-dev \
     nodejs \
-    libjemalloc1 && \
+    libjemalloc2 && \
     apt-get clean && rm -rf /var/lib/apt/lists/*
 
-ENV LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libjemalloc.so.1
+# configure jemalloc v5 with v3 behaviours (trade ram usage over performance)
+# https://twitter.com/nateberkopec/status/1442894624935137288
+# https://github.com/code-dot-org/code-dot-org/blob/5c8b24674d1c2f7e51e85dd32124e113dc423d84/cookbooks/cdo-jemalloc/attributes/default.rb#L10
+ENV MALLOC_CONF="narenas:2,background_thread:true,thp:never,dirty_decay_ms:1000,muzzy_decay_ms:0"
```

Comment:
Good catch, if that memory usage increase was the combination we might have felt it but not realized what was going on. I assume that github link was in the retweet above, but it's protected now. With that in mind, do you think it's worth preserving the context of these changed parameters somewhere else in case that github link changes or disappears?

Reply:
Sounds good to me..I dont know where though.